lib/cadet_web/views/view_helpers.ex

```diff
@@ -6,4 +6,77 @@ defmodule CadetWeb.ViewHelpers do
   def format_datetime(datetime) do
     Timex.format!(DateTime.truncate(datetime, :millisecond), "{ISO:Extended}")
   end
+
+  @doc """
+  This function allows you to build a map for a view from a map of transformations or a list of fields.
+
+  Given a `key_list`, it is the equivalent of `Map.take(source, key_list)`.
+
+  Given a map of `%{view_field: source_field, ...}`, it is the equivalent of `%{view_field: Map.get(source, source_field), ...}`
+
+  Given a map of `%{view_field: source_function, ...}`, it is the equivalent of `%{view_field: apply(source_function, source)}`
+
+  Examples:
+  ```
+  source = %{
+    foofoo: "ho",
+    barbar: "ha",
+    foobar: "hoha"
+  }
+
+  field_list = [:foofoo, :barbar]
+
+  transform_map_for_view(source, field_list)
+  > %{
+    foofoo: "ho",
+    barbar: "bar"
+  }
+
+  key_transformations = %{
+    foo: :foofoo,
+    bar: :barbar
+  }
+
+  transform_map_for_view(source, key_transformations)
+  > %{
+    foo: Map.get(source, :foofoo),
+    bar: Map.get(source, :barbar)
+  }
+
+  function_transformations = %{
+    foo: fn source -> source.foofoo <> "hoho",
```

Comment:
I think this is a better design:
```elixir
source = %{
  foofoo: "ho",
  barbar: "ha",
  foobar: "hoha"
}

function_transformations = %{
  foofoo: &(&1 <> "tst")
}

transform_map_for_view(source, functions_transformations)
# > %{
#   foofoo: "hotst"
# }
```

Not really sure why you decided to do what you did. Maybe you could elaborate?

Reply:
For me, the primary intention is to create a standardised way to transform from a `source_map` into a `view_map`. Other than the direct `source_field_name` to `view_field_name` transformation, I wanted to be able to define a `view_field` as a `fn source_map -> my_custom_transformation_to_value(source_map) end`. The transformation could involve aggregation of multiple fields from the `source_map`, calculations, etc. 

Your suggestion only applies to when `source_map` and `view_map` have the same `field_name` and you are doing a direct `value` -> `new_value` transformation.

Reply:
Ah ok fair enough.